integration_test/robots/search_robot.dart

```diff
@@ -0,0 +1,140 @@
+import 'package:collection/collection.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:tmail_ui_user/features/thread/presentation/widgets/email_tile_builder.dart';
+
+import '../base/core_robot.dart';
+
+class SearchRobot extends CoreRobot {
+  SearchRobot(super.$);
+
+  Future<void> enterQueryString(String queryString) async {
+    await $(#search_email_text_field).enterText(queryString);
+    await ensureViewVisible($(#suggestion_search_list_view));
+  }
+
+  Future<void> scrollToEndListSearchFilter() async {
+    await $.scrollUntilVisible(
+      finder: $(#mobile_sortBy_search_filter_button),
+      view: $(#search_filter_list_view),
+      scrollDirection: AxisDirection.right,
+      delta: 300,
+    );
+    await ensureViewVisible($(#mobile_sortBy_search_filter_button));
+  }
+
+  Future<void> openSortOrderBottomDialog() async {
+    await $(#mobile_sortBy_search_filter_button).tap();
+    await ensureViewVisible($(#sort_filter_context_menu));
+  }
+
+  Future<void> selectSortOrder(String sortOrderName) async {
+    await $(find.text(sortOrderName)).tap();
+    await $.pump(const Duration(seconds: 2));
+    await ensureViewVisible($(#search_email_list_notification_listener));
+  }
+
+  Future<void> expectEmailListDisplayedCorrectly({
```

Comment:
IMO, move all `expect` statement to specific scenario.

Reply:
Done